internal/tracing/handlers/handler.go

```diff
@@ -25,94 +22,37 @@ func BuildOptions(k8s client.Client, mcAddon *addonapiv1alpha1.ManagedClusterAdd
 		ClusterName:           mcAddon.Namespace,
 	}
 
-	klog.Info("Retrieving OpenTelemetry Collector template")
 	key := addon.GetObjectKey(mcAddon.Status.ConfigReferences, otelv1alpha1.GroupVersion.Group, opentelemetryCollectorResource)
 	otelCol := &otelv1alpha1.OpenTelemetryCollector{}
 	if err := k8s.Get(context.Background(), key, otelCol, &client.GetOptions{}); err != nil {
 		return resources, err
 	}
 	resources.OpenTelemetryCollector = otelCol
-	klog.Info("OpenTelemetry Collector template found")
-
-	var authCM *corev1.ConfigMap = nil
-	var caSecret *corev1.Secret = nil
-
-	for _, config := range mcAddon.Spec.Configs {
-		key := client.ObjectKey{Name: config.Name, Namespace: config.Namespace}
-		switch config.ConfigGroupResource.Resource {
-		case addon.ConfigMapResource:
-			cm := &corev1.ConfigMap{}
-			klog.Infof("processing cm %s/%s", config.Namespace, config.Name)
-			if err := k8s.Get(context.Background(), key, cm, &client.GetOptions{}); err != nil {
-				return resources, err
-			}
-
-			// Only care about cm's that configure tracing
-			if signal, ok := cm.Labels[addon.SignalLabelKey]; !ok || signal != addon.Tracing.String() {
-				klog.Info("skipped configmap")
-				continue
-			}
-
-			// If a cm doesn't have a target annotation then it's configuring authentication
-			if _, ok := cm.Annotations[manifests.AnnotationTargetOutputName]; !ok {
-				if authCM != nil {
-					klog.Warning(fmt.Sprintf("auth configmap already set to %s. new configmap %s", authCM.Name, cm.Name))
-				}
-				klog.Info("auth configmap set")
-				authCM = cm
-				continue
-			}
-
-			resources.ConfigMaps = append(resources.ConfigMaps, *cm)
-		case addon.SecretResource:
-			secret := &corev1.Secret{}
-			klog.Infof("processing secret %s/%s", config.Namespace, config.Name)
-			if err := k8s.Get(context.Background(), key, secret, &client.GetOptions{}); err != nil {
-				return resources, err
-			}
-
-			// Only care about cm's that configure tracing
-			if signal, ok := secret.Labels[addon.SignalLabelKey]; !ok || signal != addon.Tracing.String() {
-				klog.Info("skipped secret")
-				continue
-			}
+	cfg, err := otelcol.ConfigFromString(otelCol.Spec.Config)
+	if err != nil {
+		return resources, err
+	}
+	exporters, err := otelcol.GetExporters(cfg)
+	if err != nil {
+		return resources, err
+	}
 
-			// If the secret has the ca annotation then it's the secret containing the ca
-			if _, ok := secret.Annotations[authentication.AnnotationCAToInject]; ok {
-				caSecret = secret
-				continue
-			}
-		}
+	targetSecretName := make(map[authentication.Target]string)
+	for exporterName := range exporters {
+		// TODO @iblancas help!
+		targetSecretName[authentication.Target(exporterName)] = "TODO"
```

Comment:
@iblancasa will need your help migrating the tracing package to the new reconciliation. I've already started but there are some areas which I belive it would be better if they were done by someone with knowledge about the OTEL Collector API

Reply:
Should be done with [be8775f](https://github.com/rhobs/multicluster-observability-addon/pull/48/commits/be8775f19ddb26fd07b559ea918eb35f05dfa176)